main.ts

```diff
@@ -90,6 +93,11 @@ const getLatestSettings = async (
   return plugin.settings;
 };
 
+const generateToc = (content: string): string => {
+  const tocContent = toc(content).content;
+  return tocContent ? `## Table of Contents\n\n${tocContent}\n\n` : '';
```

Comment:
Why not a top-level title?

```suggestion
  return tocContent ? `# Table of Contents\n\n${tocContent}\n\n` : '';
```

Reply:
I've tweaked this now, so we will look for a `Contents` or `Table of Contents` header, and add the contents there.

Reply:
It still doesn't work. `# Contents` is not supported, but `## Contents` works.